src/readbin/readbin.ml

```diff
@@ -8,10 +8,16 @@ open Image
 module Program(Conf : Options.Provider) = struct
   open Conf
 
-  let disassemble ?(syms=Table.empty) ?roots arch mem =
-    let disasm = match roots with
-      | Some x -> disassemble ~roots:x arch mem
-      | None -> disassemble arch mem in
+  let disassemble ?img arch mem =
+    let syms = match options.symsfile with
+      | Some filename ->
+        Symtab.read ?demangle:options.demangle ~filename mem
+      | None -> match img with
+        | Some x -> Table.map (symbols x) ~f:Symbol.name
+        | None -> Table.singleton mem "bap_autogen" in
```

Comment:
can you please rename "bap_autogen" to "text"
This will result in consistent behavior, since if we don't know to what symbol address is belong, we just use "text" segment as a big symbol.